frontend/packages/kubevirt-plugin/src/components/boot-order/boot-order.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Text, TextVariants } from '@patternfly/react-core';
+import { DNDDataList, DNDDataListItem } from '../dnd-list';
+import { BootOrderEmpty } from './boot-order-empty';
+import { AddItemButton } from './add-item-button';
+import { AddItemFormSelect } from './add-item-form-select';
+import {
+  addItemMessage,
+  addItemDisabledMessage,
+  addItemSelectLabel,
+  bootOrderEmptyMessage,
+  bootOrderEmptyTitle,
+  deviceKey,
+  deviceLabel,
+  bootOrderAriaLabel,
+} from './constants';
+
+export const BootOrder = ({ devices, setDevices }: BootOrderProps) => {
+  const sources = _.sortBy(devices.filter((device) => device.value.bootOrder), 'value.bootOrder');
+  const options = devices.filter((device) => !device.value.bootOrder);
+  const [isEditMode, setEditMode] = React.useState<boolean>(false);
+
+  // Add a bootOrder to a device by key, item key is "<type>->name>".
+  const addByKey = (key: string): void => {
+    const newDevices = _.cloneDeep(devices);
+
+    const bootOrder = sources.length !== 0 ? sources[sources.length - 1].value.bootOrder + 1 : 1;
+    newDevices.find((device) => deviceKey(device) === key).value.bootOrder = bootOrder;
+
```

Comment:
\+ relax here - the array might not have been relaxed to begin with

Reply:
done